src/istio/control/http/attributes_builder.cc

```diff
@@ -62,6 +68,44 @@ void AttributesBuilder::ExtractRequestHeaderAttributes(CheckData *check_data) {
   }
 }
 
+void AttributesBuilder::SaveAuthAttributesToStruct(
```

Comment:
It is better to move this function to https://github.com/istio/proxy/blob/master/src/envoy/utils/authn.h#L29

Reply:
I tried that but it couldn't access the attributes names (See [here](https://github.com/istio/proxy/blob/624540603b7ebb206335d5cb375faa4d8ffcfbdc/src/istio/control/BUILD#L28)), I don't want to change the visibility of those attributes names so I added this in attributes_builder.cc, also it's closer to the `ExtractAuthAttributes()` function that makes it clear we are setting the right names.

Reply:
I see. Then I like to have following changes to expose attribute_names

1) move attribute_name.h to include/istio/utils/
2) create a separate library as attribute_names in src/istio/util  and set its visibility as public

Reply:
Done.